W3C.Domain/Repositories/MongoDbRepositoryBase.cs

```diff
@@ -37,15 +42,28 @@ protected Task<T> LoadFirst<T>(string id) where T : IIdentifiable
     protected Task Insert<T>(T element)
     {
         var mongoCollection = CreateCollection<T>();
-        return mongoCollection.InsertOneAsync(element);
+        var session = _transactionCoordinator?.GetCurrentSession();
+        return session != null
+            ? mongoCollection.InsertOneAsync(session, element)
+            : mongoCollection.InsertOneAsync(element);
     }
 
     protected async Task<List<T>> LoadAll<T>(Expression<Func<T, bool>> expression = null, int? limit = null)
     {
-        if (expression == null) expression = l => true;
+        expression ??= l => true;
```

Comment:
If there's no expression, it's probably better to just find with an empty filter instead? It's at least more readable and may be handled by mongo better?

```csharp
var filter = Builders<T>.Filter.Empty;
mongoCollection.Find(filter);
```

Reply:
Good call, thanks!